src/parser/parser.yy

```diff
@@ -2862,22 +2888,46 @@ create_space_sentence
         auto sentence = new CreateSpaceSentence($4, $3);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_COMMENT ASSIGN STRING {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setComment($7);
+        $$ = sentence;
+    }
     | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON name_label {
         auto sentence = new CreateSpaceSentence($4, $3);
         sentence->setGroupName($6);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label KW_ON name_label KW_COMMENT ASSIGN STRING {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setGroupName($6);
+        sentence->setComment($9);
+        $$ = sentence;
+    }
     | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN {
         auto sentence = new CreateSpaceSentence($4, $3);
         sentence->setOpts($6);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN KW_COMMENT ASSIGN STRING {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setOpts($6);
+        sentence->setComment($10);
+        $$ = sentence;
+    }
     | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN KW_ON name_label {
         auto sentence = new CreateSpaceSentence($4, $3);
         sentence->setGroupName($9);
         sentence->setOpts($6);
         $$ = sentence;
     }
+    | KW_CREATE KW_SPACE opt_if_not_exists name_label L_PAREN space_opt_list R_PAREN KW_ON name_label KW_COMMENT ASSIGN STRING {
+        auto sentence = new CreateSpaceSentence($4, $3);
+        sentence->setGroupName($9);
+        sentence->setOpts($6);
+        sentence->setComment($12);
+        $$ = sentence;
+    }
```

Comment:
Maybe you need to refactor the `create_space_sentence` syntax implementation:

```cpp
create_space_without_group_sentence
  : ... {
  }
  ;
create_space_without_comment_sentence
  : create_space_without_group_sentence {
    $$ = $1;
  }
  | create_space_without_group_sentence KW_ON name_label {
    $1->setGroup($3);
    $$ = $1;
  }
  ;
create_space_sentence
  : create_space_without_comment_sentence {
     $$ = $1;
  }
  | create_space_without_comment_sentence KW_COMMENT ASSIGN STRING {
    $1->setComment($4);
    $$ = $1;
  }
  ;

```

Reply:
This should refactor in a separate PR.